include/tc/core/cuda/cuda_execution_engine.h

```diff
@@ -31,64 +32,54 @@ namespace tc {
 /// able to execute the kernels for multiple TC i.e. given the language which
 /// can have multiple TCs, people should be able to run things by just calling
 /// out the run function with the name of function and the inputs to run on.
-class CudaExecutionEngine {
+class CudaExecutionEngine : public ExecutionEngine {
  public:
-  struct ExecutorInfo {
-    ExecutorInfo(
+  struct CudaExecutorInfo : public ExecutionEngine::ExecutorInfo {
+    CudaExecutorInfo(
         std::string id,
         std::vector<const DLTensor*> inputsInfo,
-        std::unique_ptr<MappingOptions> options,
+        const MappingOptions& options,
         lang::TreeRef tc,
         size_t handle)
-        : identifier(id),
-          inputsInfo(dlutils::makeDLTensorVector(inputsInfo)),
-          options(std::move(options)),
-          exec(tc, inputsInfo),
-          objectLocalHandle(handle) {}
+        : ExecutionEngine::ExecutorInfo(
+              id,
+              inputsInfo,
+              options.toProtobufSerializedString(),
+              tc,
+              handle) {
+      exec =
+          std::unique_ptr<CudaTcExecutor>(new CudaTcExecutor(tc, inputsInfo));
```

Comment:
Why are you overwriting this field?
Why not pass it to the constructor of `ExecutorInfo`?